datalake/src/main/java/com/sequenceiq/datalake/service/sdx/status/SdxStatusService.java

```diff
@@ -53,11 +58,46 @@
     @Inject
     private Clock clock;
 
+    @Inject
+    private FlowLogService flowLogService;
+
+    @Inject
+    private FlowChainLogService flowChainLogService;
+
     public void setStatusForDatalakeAndNotify(DatalakeStatusEnum status, String statusReason, SdxCluster sdxCluster) {
         setStatusForDatalake(status, statusReason, sdxCluster);
         sdxNotificationService.send(status.getDefaultResourceEvent(), sdxCluster);
     }
 
+    public void setStatusForDatalakeAndNotify(DatalakeStatusEnum status, String statusReason, SdxCluster sdxCluster, long flowId) {
+        FlowLog flowLog = flowLogService.getLastFlowLog(flowId).get();
+        setStatusForDatalake(status, statusReason, sdxCluster);
+        FlowChainLog flowChainLog = flowChainLogService.findFirstByFlowChainIdOrderByCreatedDesc(flowLog.getFlowChainId()).get();
+        if (!(flowChainLog.getFlowChainType().equals(DatalakeResizeFlowEventChainFactory.class.getName()) &&
```

Comment:
The methods name says nothing about the fact that if this `!(flowChainLog.getFlowChainType().equals(DatalakeResizeFlowEventChainFactory.class.getName())` condition is not true than we don't send notification. By the way I wouldn't include such logic in this class.